From ab45cc04c52ed6c381676dd911b61dbdaa1678a1 Mon Sep 17 00:00:00 2001
From: Nobuhiro Iwamatsu <nobuhiro.iwamatsu@miraclelinux.com>
Date: Wed, 13 Feb 2019 13:16:26 +0900
Subject: [PATCH 3/5] Upstream-Status: Inappropriate [OE specific]

commonio.c: fix unexpected open failure in chroot environment

When using commands with '-R <newroot>' option in our pseudo environment,
we would usually get the 'Pemission Denied' error. This patch serves as
a workaround to this problem.

Note that this patch doesn't change the logic in the code, it just expands
the codes.

Signed-off-by: Chen Qi <Qi.Chen@windriver.com>
Signed-off-by: Nobuhiro Iwamatsu <nobuhiro.iwamatsu@miraclelinux.com>
---
 lib/commonio.c | 16 ++++++++++++----
 1 file changed, 12 insertions(+), 4 deletions(-)

diff --git a/lib/commonio.c b/lib/commonio.c
index bfae8be..ffefc51 100644
--- a/lib/commonio.c
+++ b/lib/commonio.c
@@ -611,10 +611,18 @@ int commonio_open (struct commonio_db *db, int mode)
 	db->cursor = NULL;
 	db->changed = false;
 
-	fd = open (db->filename,
-	             (db->readonly ? O_RDONLY : O_RDWR)
-	           | O_NOCTTY | O_NONBLOCK | O_NOFOLLOW);
-	saved_errno = errno;
+	if (db->readonly) {
+		fd = open (db->filename,
+			   (true ? O_RDONLY : O_RDWR)
+			   | O_NOCTTY | O_NONBLOCK | O_NOFOLLOW);
+		saved_errno = errno;
+	} else {
+		fd = open (db->filename,
+			   (false ? O_RDONLY : O_RDWR)
+			   | O_NOCTTY | O_NONBLOCK | O_NOFOLLOW);
+		saved_errno = errno;
+	}
+
 	db->fp = NULL;
 	if (fd >= 0) {
 #ifdef WITH_TCB
-- 
2.20.1

